tutorials/singularity-containers-with-cloud-build/README.md

```diff
@@ -0,0 +1,317 @@
+see the [documentation here](https://cloud.google.com/resource-manager/docs/creating-managing-organization)).
+
+Use these ```gcloud``` commands to retrieve the values required by ```env.sh```.
+
+```bash
+# List organizations - you can use the name or the id
+gcloud organizations list
+
+# List billing accounts
+gcloud alpha billing accounts list
+
+# List projects
+gcloud projects list
+
+# List zones
+gcloud compute zones list
+
+```
+
+If you are creating a new project use the name you intend to give it.
+
+Source ```env.sh``` to define the environment variables in your current shell session.
+
+```bash
+source ./env.sh
+```
+
+### (Optional) Create a new project
+
+To use a brand new project rather than an existing one, create the new project with the commands:
+
+```bash
+gcloud projects create $PROJECT --organization=$ORG
+gcloud beta billing projects link $PROJECT --billing-account=$(gcloud beta billing accounts list | grep $BILLING_ACCOUNT | awk '{print $1}')
+gcloud config configurations create -- activate $PROJECT
+```
+
+### Use the desired project and zone
+
+```bash
+gcloud config set core/project $PROJECT
+gcloud config set compute/zone $ZONE
+```
+
+### Enable the necessary Cloud Platform APIs
+
+The following commands will enable Google Compute Engine, Google Cloud Builder,
+and Google Container Registry:
+
+```bash
+gcloud services enable compute.googleapis.com
+gcloud services enable cloudbuild.googleapis.com
+gcloud services enable containerregistry.googleapis.com
+```
+
+## Build a Singularity container
+
+Cloud Build supports the definition and use of _custom build steps_ to extend the range of tasks it can handle. In the first step of this section
+you create a Singularity custom build step. In the third step you use the Singularity custom build step to build a Singularity container and save
+it in Cloud Storage. Once the Singuarlity custom build step has been created in your project you can use it to create as many Singularity containers
+as you like.
+
+### 1. Create a Singularity builder
+
+The ```singularity-buildstep``` directory contains the files you use to create the Singularity custom build step.
+
+* ```builder.yaml``` is the recipe to create the Singularity builder. Specifically, it's a configuration file you will submit to build the custom build step
+* ```Dockerfile``` contains the steps required to download and build Singularity in a ```go``` container
+
+Move to that directory
+
+```bash
+cd singularity-buildstep
+```
+
+Then use this command to build the custom build step.
+
+```bash
+gcloud builds submit --config=builder.yaml --substitutions=_SINGULARITY_VERSION="3.0.2" .
+```
+
+At the time of this writing the latest stable version of Singularity is 3.0.2. To use a different, possibly newer, version modify the value of the
+_SINGULARITY_VERSION substitution accordingly. What does this verison coincide with? It should match a [release tag](https://github.com/sylabs/singularity/releases)
```

Comment:
Need to escape the underscore at beginning with a backslash: \_SINGU...

Reply:
Done